src/layout/scrolling.rs

```diff
@@ -1749,6 +1759,123 @@ impl<W: LayoutElement> ScrollingSpace<W> {
+        {
+            return self.move_column_to(target_column_idx);
+        }
+
+        let source_tile_idx = self.columns[source_column_idx].active_tile_idx;
+        let target_tile_idx = self.columns[target_column_idx].active_tile_idx;
+        let source_column_drained = self.columns[source_column_idx].tiles.len() == 1;
+
+        // capture the original positions of the tiles
+        let (source_pt, target_pt) = (
+            Point::from((self.column_x(source_column_idx), 0.))
+                + self.columns[source_column_idx].tile_offset(source_tile_idx),
+            Point::from((self.column_x(target_column_idx), 0.))
+                + self.columns[target_column_idx].tile_offset(target_tile_idx),
+        );
+
+        let transaction = Transaction::new();
+
+        // If the source column contains a single tile, this will also remove the column.
+        // When this happens `source_column_drained` will be set and the column will need to be
+        // recreated with `add_tile`
+        let source_removed = self.remove_tile_by_idx(
+            source_column_idx,
+            source_tile_idx,
+            transaction.clone(),
+            None,
+        );
+
+        {
+            // special case when the source column disappears after removing its last tile
+            let adjusted_target_column_idx =
+                if direction == ScrollDirection::Right && source_column_drained {
+                    target_column_idx - 1
+                } else {
+                    target_column_idx
+                };
+
+            self.add_tile_to_column(
+                adjusted_target_column_idx,
+                Some(target_tile_idx),
+                source_removed.tile,
+                false,
+            );
+
+            let RemovedTile {
+                tile: target_tile, ..
+            } = self.remove_tile_by_idx(
+                adjusted_target_column_idx,
+                target_tile_idx + 1,
+                transaction.clone(),
+                None,
+            );
+
+            if source_column_drained {
+                // recreate the drained column with only the target tile
+                self.add_tile(
+                    Some(source_column_idx),
+                    target_tile,
+                    true,
+                    source_removed.width,
+                    source_removed.is_full_width,
+                    None,
+                )
+            } else {
+                // simply add the removed target tile to the source column
+                self.add_tile_to_column(
+                    source_column_idx,
+                    Some(source_tile_idx),
+                    target_tile,
+                    false,
+                );
+            }
+        }
+
+        // update the active tile in the modified columns
+        self.columns[source_column_idx].active_tile_idx = source_tile_idx;
+        self.columns[target_column_idx].active_tile_idx = target_tile_idx;
+
+        // Animations
```

Comment:
The animation for the target tile is a little broken: it jumps vertically. This should be fixed.